sway/tree/view.c

```diff
@@ -940,12 +943,16 @@ void view_center_and_clip_surface(struct sway_view *view) {
 
 	// only make sure to clip the content if there is content to clip
 	if (!wl_list_empty(&con->view->content_tree->children)) {
-		wlr_scene_subsurface_tree_set_clip(&con->view->content_tree->node, &(struct wlr_box){
-			.x = con->view->geometry.x,
-			.y = con->view->geometry.y,
-			.width = con->current.content_width,
-			.height = con->current.content_height,
-		});
+		struct wlr_box clip = {0};
+		if (clip_to_geometry) {
+			clip = (struct wlr_box){
+				.x = con->view->geometry.x,
```

Comment:
I think it should just be using `con->current.content_x` instead of `con->view->geometry.x` here.

Reply:
Tried this but it did not work properly. Looks like `content_x` and `content_y` are absolute coordinates, while `subsurface_tree_set_clip` expects surface-local coordinates.